packages/router/history.ts

```diff
@@ -615,8 +618,8 @@ function getUrlBasedHistory(
     if (nextIndex != null) {
       let delta = nextIndex - index;
       action = nextAction;
+      index = nextIndex;
       if (listener) {
-        index = nextIndex;
```

Comment:
We should always be updating this even if no one is listening - never showed up as an issue since we always have a `listener`